core/modules/ccontrol/queryMsg.cc

```diff
@@ -43,19 +43,23 @@ int queryMsgGetCount(int session) {
 }
 
 // Python call: msg, chunkId, code, timestamp = queryMsgGetMsg(session, idx)
-std::string queryMsgGetMsg(int session, int idx, int* chunkId, int* code, time_t* timestamp) {
+std::string queryMsgGetMsg(int session, int idx, int* chunkId, int* code, std::string* severity, time_t* timestamp) {
 
     qdisp::QueryMessage msg = UserQuery_get(session).getMessageStore()->getMessage(idx);
     *chunkId = msg.chunkId;
     *code = msg.code;
     *timestamp = msg.timestamp;
+    *severity = qdisp::to_string(msg.severity);
```

Comment:
Actually, for output parameters, pointers are generally preferred as they make clear that the parameter is different from others.  Another option is to package all the outputs into a struct or class or `pair<>` that can be returned.